### PR TITLE
Create and reuse gb-tree-search component

### PR DESCRIPTION
**Pull request by forus**
**Note number of expended nodes is not controlled yet.**
It's expected that strategy to fix performance issue will change.

### CODE REVIEW

File: src/app/modules/gb-side-panel-module/accordion-components/gb-tree-nodes/gb-tree-nodes.component.ts [hunk 1]
```diff
@@ -39,19 +39,10 @@ export class GbTreeNodesComponent implements OnInit, AfterViewInit, AfterViewChe
...
-  // the search term in the text input box to filter the tree
-  searchTerm = '';
-  // the delay before triggering updating methods
-  // as the PrimeNg tree nodes reconstruct DOM nodes and css styles when data changes,
-  // and this will take a while
-  delay: number;
   // indicate if the initUpdate is finished
   initUpdated: boolean;
+
   // max number of expanded nodes in search
```
Comment (ewelinagr) [inline src/app/modules/gb-side-panel-module/accordion-components/gb-tree-nodes/gb-tree-nodes.component.ts]: looks like this comment should be out too

File: src/app/modules/gb-side-panel-module/accordion-components/gb-variables/gb-variables.component.css [hunk 2]
```diff
@@ -20,3 +20,26 @@
...
+:host >>> .gb-tree-filter-input > input {
+  font-size: small !important;
+  height: 1.8em;
+  width: 100%;
+  min-width: 3em;
+  max-width: 7em;
+}
+
+/*tree node styles for filtering*/
+:host >>> .gb-highlight-treenode {
```
Comment (ewelinagr) [inline src/app/modules/gb-side-panel-module/accordion-components/gb-variables/gb-variables.component.css]: looks like a duplication - this class is already defined in gb-side-panel.component.css

File: src/app/modules/gb-side-panel-module/accordion-components/gb-variables/gb-variables.component.css [hunk 3]
```diff
@@ -20,3 +20,26 @@
...
+  background-color: var(--gb-orange-light) !important;
+  border-radius: 10px !important;
+  font-weight: bold !important;
+}
+
+.filter-btn {
```
Comment (ewelinagr) [inline src/app/modules/gb-side-panel-module/accordion-components/gb-variables/gb-variables.component.css]: duplicated in gb-tree-search.component.css?

File: src/app/modules/gb-side-panel-module/accordion-components/gb-variables/gb-variables.component.css [hunk 4]
```diff
@@ -20,3 +20,26 @@
   margin-top: 0.5em;
   font-size: small;
 }
+
+/*Style for the search box in the tree-node accordion on the left side*/
+:host >>> .gb-tree-filter-input > input {
```
Comment (ewelinagr) [inline src/app/modules/gb-side-panel-module/accordion-components/gb-variables/gb-variables.component.css]: same here, duplicated in gb-tree-search.component.css?

File: src/app/modules/gb-side-panel-module/accordion-components/gb-tree-nodes/gb-tree-nodes.component.html [hunk 5]
```diff
@@ -1,22 +1,9 @@
...
-              (click)="clearFilter()"
-              type="button">clear
-      </button>
-      <div *ngIf="hits > 0" class="filter-result">
-        {{hits}} results found
-      </div>
-    </div>
+    <gb-tree-search
+      [tree]="treeNodeService.treeNodes"
+    ></gb-tree-search>
```
Comment (beaugogh) [inline src/app/modules/gb-side-panel-module/accordion-components/gb-tree-nodes/gb-tree-nodes.component.html]: I like the idea of abstracting it only for tree nodes, currently nicely separated from the category list, but also allow for future inclusion :)

File: src/app/modules/gb-side-panel-module/accordion-components/gb-tree-nodes/gb-tree-nodes.component.ts [hunk 6]
```diff
@@ -184,129 +170,6 @@ export class GbTreeNodesComponent implements OnInit, AfterViewInit, AfterViewChe
     }
   }
 
-  /**
```
Comment (beaugogh) [inline src/app/modules/gb-side-panel-module/accordion-components/gb-tree-nodes/gb-tree-nodes.component.ts]: just wondering are the functions such as removeFalsePrimeNgClasses, clearFilter completely removed or they are reused somehow in the tree-search component?

File: src/app/modules/gb-side-panel-module/accordion-components/gb-variables/gb-variables.component.css (hunk 4 above)
Comment (beaugogh) [reply src/app/modules/gb-side-panel-module/accordion-components/gb-variables/gb-variables.component.css]: ideally we want to confine the css the its component, in this case, the styles for gb-tree-search component will be in the gb-tree-search.css

File: src/app/modules/gb-side-panel-module/accordion-components/gb-variables/gb-variables.component.css (hunk 2 above)
Comment (beaugogh) [reply src/app/modules/gb-side-panel-module/accordion-components/gb-variables/gb-variables.component.css]: move tree css to the tree search component folder

File: src/app/modules/gb-side-panel-module/gb-side-panel.component.css [hunk 7]
```diff
@@ -46,3 +46,10 @@
 :host >>> .ui-accordion .ui-accordion-header {
   background-color: white;
 }
+
+/*tree node style for highlighted node*/
+:host >>> .gb-highlight-treenode {
```
Comment (beaugogh) [inline src/app/modules/gb-side-panel-module/gb-side-panel.component.css]: move tree-related css to the tree search component

**Pull request merged**